wombase_backend/server/apps/staff/views.py

```diff
@@ -1,12 +1,47 @@
-from rest_framework import generics
+from rest_framework import generics, status, viewsets
+from rest_framework.response import Response
+
+from .models import Employee, EmployeeRole
+from .serializers import EmployeeDetailsSerializer, EmployeeListCreateSerializer, EmployeeRoleSerializer
 
-from .models import Employee
-from .serializers import EmployeeSerializer
 
 class EmployeeListCreateAPIView(generics.ListCreateAPIView):
     queryset = Employee.objects.all()
-    serializer_class = EmployeeSerializer
+    serializer_class = EmployeeListCreateSerializer
+
 
 class EmployeeRetrieveUpdateDestroyAPIView(generics.RetrieveUpdateDestroyAPIView):
     queryset = Employee.objects.all()
-    serializer_class = EmployeeSerializer
+    serializer_class = EmployeeDetailsSerializer
+
+    def patch(self, request, *args, **kwargs):
+        user_id = kwargs.get('pk')
+        user = Employee.objects.get(id=user_id)
+
+        phone_number = request.data.get('phone_number', user.phone_number)
+        first_name = request.data.get('first_name', user.first_name)
+        last_name = request.data.get('last_name', user.last_name)
+        email = request.data.get('email', user.email)
+        new_password = request.data.get('new_password')
+        role = EmployeeRole.objects.get(user.role)
+
+        user.phone_number = phone_number
+        user.first_name = first_name
+        user.last_name = last_name
+        user.email = email
+        user.role = role
```

Comment:
a lot of unnecessay code. Iterate over request.data.items() and use setattr for user